File: go/fixture/go-mini/internal/models/device.go
// Package models holds the domain types shared by every layer of the service.
package models

import "time"

// Device is a device.
type Device struct {
	ID       string
	Tenant   string
	Status   string
	Version  string
	Tags     []string
	LastSeen time.Time
}

// SetStatus sets the status.
func (d *Device) SetStatus(s string) {
	d.Status = s
}

// IsOnline returns whether the device is online.
func (d Device) IsOnline() bool {
	return d.Status == "READY" || d.Status == "DEGRADED"
}

File: py/fixture/py-mini/internal/models/device.py
"""Devices as the fleet knows them."""

from dataclasses import dataclass, field
from datetime import UTC, datetime

# The zero time: a device that has never been seen.
NEVER = datetime.min.replace(tzinfo=UTC)


@dataclass
class Device:
    """Device is a device."""

    id: str
    tenant: str
    status: str
    version: str = ""
    tags: list[str] = field(default_factory=list)
    last_seen: datetime = NEVER

    def set_status(self, s: str) -> None:
        """Set the status."""
        self.status = s

    def is_online(self) -> bool:
        """Return whether the device is online."""
        return self.status in ("READY", "DEGRADED")
